include/openPMD/IO/ADIOS/ADIOS2IOHandler.hpp

```diff
@@ -579,8 +580,36 @@ namespace detail
          */
         void drop( );
 
-        std::map< std::string, adios2::Params >
-        availableAttributesTemporary( std::string const & variable );
+        AttributeMap_t const &
+        availableAttributes();
+
+        std::vector< std::string >
+        availableAttributesPrefixed( std::string const & prefix );
+
+        void
+        invalidateAttributesMap();
+
+        AttributeMap_t const &
+        availableVariables();
+
+        std::vector< std::string >
+        availableVariablesPrefixed( std::string const & prefix );
+
+        void
+        invalidateVariablesMap();
+
+    private:
+        /*
```

Comment:
```suggestion
        /** @todo <Title of the nature of the work-around>
```
so we can find this again.